models/product-data/src/generator.ts

```diff
@@ -0,0 +1,38 @@
+import * as Category from '@commercetools-test-data/category';
+import { LocalizedString, Reference } from '@commercetools-test-data/commons';
+import { fake, Generator } from '@commercetools-test-data/core';
+import { TProductData } from './types';
+
+// https://docs.commercetools.com/api/projects/products#productdata
+
+const generator = Generator<TProductData>({
+  fields: {
+    name: fake(() => LocalizedString.random()),
+    categories: fake(() => [Category.random()]),
+    categoryOrderHints: fake((f) => ({
+      [f.datatype.uuid()]: String(Math.random()),
+    })),
+    description: fake(() => LocalizedString.random()),
+    slug: fake(() => LocalizedString.presets.ofSlugs()),
+    metaTitle: fake(() => LocalizedString.random()),
+    metaDescription: fake(() => LocalizedString.random()),
+    metaKeywords: fake(() => LocalizedString.random()),
+    // TODO: Include random ProductVariant when available
+    masterVariant: null,
+    // TODO: Include random ProductVariant when available
+    variant: null,
+    // TODO: Include random ProductVariant[] when available
+    variants: [],
+    // TODO: Include random ProductVariant[] when available
+    allVariants: [],
+    // TODO: Include random SearchKeywords when available
+    searchKeywords: null,
+    // TODO: Include random SearchKeywords[] when available
+    searchKeyword: [],
+    categoriesRef: fake(() => [Reference.presets.category()]),
```

Comment:
Should this field be built in the transformed fields based on the values in `categories`? If not, they won't match. Perhaps we don't care.

Reply:
Great thinking, decided to care 🤣 See 1e52392ab548a1ebf0f65ef079c589af9ded57cf